File: src/test/java/com/raychenon/leetcode/SingleNumberTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 11/4/2020
 * https://leetcode.com/problems/single-number/
 */
public class SingleNumberTest {

    private static SingleNumber singleNumber;

    @BeforeClass
    public static void setUp() {
        singleNumber = new SingleNumber();
    }

    @Test
    public void example1Test() {
        int[] array = {2, 2, 1};
        assertEquals(1, singleNumber.singleNumber(array));
    }

    @Test
    public void example2Test() {
        int[] array = {4, 1, 2, 1, 2};
        assertEquals(4, singleNumber.singleNumber(array));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/SingleNumberTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 11/4/2020
 * https://leetcode.com/problems/single-number/
 */
class SingleNumberTest {

    @Test
    fun example1Test() {
        Assert.assertEquals(1, SingleNumber.singleNumber(intArrayOf(2, 2, 1)))
    }

    @Test
    fun example2Test() {
        Assert.assertEquals(4, SingleNumber.singleNumber(intArrayOf(4, 1, 2, 1, 2)))
    }

}